.github/workflows/generate-python-classes.yml

```diff
@@ -0,0 +1,219 @@
+        if [ "$PR_REPO" != "${{ github.repository }}" ]; then
+          echo "⚠️ This is a fork PR. Auto-commit requires a GitHub App or PAT token."
+          echo "Please run 'make modules-gen' locally and push the changes."
+          exit 0
+        fi
+        
+        # Add generated Python classes
+        git add modules/*/src/htan_*/datamodel/*.py || true
+        
+        # Commit changes (skip if no changes or commit already exists)
+        git commit -m "chore: auto-generate Python classes from schema changes
+
+        Auto-generated Python classes from LinkML schema updates.
+        
+        **Auto-generated by GitHub Actions workflow**
+        [skip ci]" || echo "No changes to commit or commit already exists"
+        
+        # Push to PR branch (GITHUB_TOKEN works for same-repo PRs)
+        git push origin "HEAD:$PR_BRANCH" || echo "Push failed"
+    
+    - name: Fail if Python classes are outdated (push to main only)
+      if: steps.check-changes.outputs.has-changes == 'true' && github.event_name == 'push'
+      run: |
+        echo "❌ **DEPLOYMENT CHECK FAILED**"
+        echo ""
+        echo "The Python classes are not up to date with the current schemas."
+        echo "This means the generated Python code doesn't match the LinkML YAML files."
+        echo ""
+        echo "**To fix this:**"
+        echo "1. Run the manual Python generation workflow"
+        echo "2. Or run locally: \`make modules-gen\`"
+        echo "3. Commit the updated Python classes"
+        echo ""
+        echo "**Changed files:**"
+        git diff --name-only
+        exit 1
+    
+    - name: Create Pull Request (Push to main only)
+      if: steps.check-changes.outputs.has-changes == 'true' && github.event_name == 'push'
+      env:
+        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        # Configure git
+        git config --local user.email "action@github.com"
+        git config --local user.name "GitHub Action"
+        
+        # Create a new branch for this update
+        branch_name="update-python-classes-$(date +%Y%m%d-%H%M%S)"
+        git checkout -b "$branch_name"
+        
+        # Add all changes
+        git add modules/*/src/htan_*/datamodel/*.py
+        
+        # Commit changes
+        git commit -m "feat: update Python classes from schema changes
+
+        Auto-generated Python classes from LinkML schema updates.
+        
+        ### Updated Modules:
+        - Biospecimen: Generated from domains/biospecimen.yaml
+        - Clinical: Generated from domains/clinical.yaml  
+        - WES: Generated from domains/level_1.yaml
+        
+        ### Changes:
+        - Updated datamodel classes
+        - Regenerated schema_classes.py files
+        - All tests passing
+        
+        **Auto-generated by Python class generation workflow**"
+        
+        # Push branch
+        git push origin "$branch_name"
+        
+        # Create pull request
```

Comment:
Can you clarify this? I thought your proposed solution was to just push to the same branch, unless I'm reading the description wrong. 

This is only done on push to main?